l10n_es_aeat/models/l10n_es_aeat_report_tax_mapping.py

```diff
@@ -0,0 +1,194 @@
+            # Buscar configuración de mapeo de impuestos
+            tax_code_map = self.env["l10n.es.aeat.map.tax"].search(
+                [
+                    ("model", "=", report.number),
+                    "|",
+                    ("date_from", "<=", report.date_start),
+                    ("date_from", "=", False),
+                    "|",
+                    ("date_to", ">=", report.date_end),
+                    ("date_to", "=", False),
+                ],
+                limit=1,
+            )
+            if tax_code_map:
+                tax_lines = []
+                for map_line in tax_code_map.map_line_ids:
+                    tax_lines.append(report._prepare_tax_line_vals(map_line))
+                report.tax_line_ids = [(0, 0, x) for x in tax_lines]
+        return res
+
+    def unlink(self):
+        self.mapped("tax_line_ids").unlink()
+        return super().unlink()
+
+    def _prepare_tax_line_vals(self, map_line):
+        self.ensure_one()
+        move_lines = self._get_tax_lines(self.date_start, self.date_end, map_line)
+        if map_line.sum_type == "credit":
+            amount = sum(move_lines.mapped("credit"))
+        elif map_line.sum_type == "debit":
+            amount = sum(move_lines.mapped("debit"))
+        else:  # map_line.sum_type == 'both'
+            amount = sum(move_lines.mapped("credit")) - sum(move_lines.mapped("debit"))
+        if map_line.inverse:
+            amount = (-1.0) * amount
+        return {
+            "model": self._name,
+            "res_id": self.id,
+            "map_line_id": map_line.id,
+            "amount": amount,
+            "move_line_ids": [(6, 0, move_lines.ids)],
+        }
+
+    def _get_partner_domain(self):
+        return []
+
+    def _get_move_line_domain(self, date_start, date_end, map_line):
+        self.ensure_one()
+        taxes = self.get_taxes_from_templates(map_line.tax_ids)
+        move_line_domain = [
+            ("company_id", "child_of", self.company_id.id),
+            ("date", ">=", date_start),
+            ("date", "<=", date_end),
+            ("parent_state", "=", "posted"),
+        ]
+        if map_line.move_type == "regular":
+            move_line_domain.append(
+                ("move_id.move_type", "in", ("receivable", "payable", "liquidity"))
+            )
+        elif map_line.move_type == "refund":
+            move_line_domain.append(
+                ("move_id.move_type", "in", ("receivable_refund", "payable_refund"))
+            )
+        if map_line.field_type == "base":
+            move_line_domain.append(("tax_ids", "in", taxes.ids))
+        elif map_line.field_type == "amount":
+            move_line_domain.append(("tax_line_id", "in", taxes.ids))
+        else:  # map_line.field_type == 'both'
+            move_line_domain += [
+                "|",
+                ("tax_line_id", "in", taxes.ids),
+                ("tax_ids", "in", taxes.ids),
+            ]
```

Comment:
@MiquelRForgeFlow el nuevo cambio sería,
if map_line.campo_many2many_nuevo:
move_line_domain.append(("campo_cuenta", "in", mape_line.campo_many2many_nuevo.ids))